plotly/tools.py

```diff
@@ -1503,35 +1503,53 @@ def _unconvert_from_RGB_255(colors):
             return un_rgb_colors
 
     @staticmethod
-    def _map_array2color(array, colormap, vmin, vmax):
+    def _map_face2color(face, colormap, vmin, vmax):
         """
-        Normalize values in array by vmin/vmax and return plotly color strings.
+        Normalize facecolor values by vmin/vmax and return rgb-color strings
 
-        This function takes an array of values along with a colormap and a
-        minimum (vmin) and maximum (vmax) range of possible z values for the
-        given parametrized surface. It returns an rgb color based on the
-        relative position of zval between vmin and vmax
+        This function takes a tuple color with elements between 0 and 1, along
+        with a colormap and a minimum (vmin) and maximum (vmax) range of
+        possible mean distances for the given parametrized surface. It returns
+        an rgb color based on the mean distance between vmin and vmax
 
         """
         if vmin >= vmax:
             raise exceptions.PlotlyError("Incorrect relation between vmin "
                                          "and vmax. The vmin value cannot be "
                                          "bigger than or equal to the value "
                                          "of vmax.")
-        # find distance t of zval from vmin to vmax where the distance
-        # is normalized to be between 0 and 1
-        t = (array - vmin) / float((vmax - vmin))
-        t_colors = FigureFactory._find_intermediate_color(colormap[0],
-                                                          colormap[1],
-                                                          t)
-        t_colors = t_colors * 255.
-        labelled_colors = ['rgb(%s, %s, %s)' % (i, j, k)
-                           for i, j, k in t_colors.T]
-        return labelled_colors
+        # find the normalized distance t of a triangle face between vmin and
+        #vmax where the distance is normalized between 0 and 1
+        t = (face - vmin) / float((vmax - vmin))
+
+        if len(colormap) <= 1:
+            t_color = colormap[0]
+            color = FigureFactory._convert_to_RGB_255(t_color)
+            color = FigureFactory._label_rgb(color)
+        else:
+            # account for colormaps with more than one color
+            incr = 1./(len(colormap) - 1)
+            low_color_index = int(t/incr)
+
+            if t == 1:
+                t_color = colormap[low_color_index]
+                color = FigureFactory._convert_to_RGB_255(t_color)
+                color = FigureFactory._label_rgb(color)
+
+            else:
+                t_color = FigureFactory._find_intermediate_color(
```

Comment:
Also, just my intuition here, I think it's better to have this handled by the colormap itself, rather than internally in a function like this. E.g., a matplotlib colormap will accept any value from 0 to 1, and will linearly interpolate according to the colormap's parameters. Is there some kind of functionality like this in plotly? (this PR is implementing something like this, but IMO it should be a more general function that is used elsewhere in the codebase and not just for trisurf)